packages/datepicker-react/src/DatePicker.tsx

```diff
@@ -14,57 +14,91 @@ interface Props {
     label?: string;
     monthLabel?: string;
     yearLabel?: string;
+    placeholder?: string;
     months?: string[];
     days?: string[];
     initialDate?: Date;
     onChange?: (date: Date) => void;
-    onlyFuture?: boolean;
     initialShow?: boolean;
     className?: string;
+    helpLabel?: string;
+    errorLabel?: string;
 }
 
+const dayMonthYearRegex = /^(\d\d)\.(\d\d)\.(\d{4})/;
+
 export function DatePicker({
     label = "Velg dato",
     monthLabel = "Måned",
     yearLabel = "År",
+    placeholder = "dd.mm.åååå",
     months,
     days,
-    initialDate = new Date(),
+    initialDate,
     onChange,
-    onlyFuture = true,
     initialShow = false,
     className = "",
+    helpLabel,
+    errorLabel,
 }: Props) {
-    const [today] = useState(Date.now() - (Date.now() % 864e3));
     const [date, setDate] = useState(initialDate);
-    const [showPicker, togglePicker] = useState(initialShow);
+    const [datepickerHidden, setDatepickerHidden] = useState(!initialShow);
+    const [dateString, setDateString] = useState(initialDate ? formatDate(initialDate) : "");
+    const openDatepicker = () => setDatepickerHidden(false);
+    const closeDatepicker = () => setDatepickerHidden(true);
+
+    const datepickerRef = useRef<HTMLDivElement>(null);
+    useClickOutside(datepickerRef, closeDatepicker);
```

Comment:
Cool! I like this a lot more. I think we should consider creating a compound hook with all of these if we see more components that has the same needs. 

Functional and beautiful!